### ISSUE TITLE
Numeric UDA values above 2,147,483,647 overflow without warning

### ISSUE THREAD

**Issue by meonkeys** [GothenburgBitFactory/taskwarrior#2101]
I created a numeric [UDA](https://taskwarrior.org/docs/udas.html) as follows:

```
uda.rtmid.type=numeric
uda.rtmid.label=RTMid
```

An example/typical `rtmid` value for a task is `221330186`. I use RTMid to store IDs from tasks [imported](https://gitlab.com/meonkeys/migrate-rtm-to-tw) into Taskwarrior from Remember the Milk.

Unexpectedly (to me), I'm unable to use the filter `rtmid:221330186` to match a task with that "exact" UDA key and value. I gather from #1356 that numeric UDAs are handled in memory as floating point values, so maybe that's why it is failing. I haven't reviewed the code.

If I change the type to `string`, everything works as I'd expect.

## Repro

1. Create a numeric UDA named `rtmid`.
1. Add a task: `task buy a phone rtmid:221330186`.
1. Try to filter for this task `task rtmid:221330186 list`.

Expected output: one task; the one we just added.

Actual output: "No matches."

## `task diag` output

```
task 2.5.1
   Platform: Linux

Compiler
    Version: 7.3.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Jan  9 2019 07:55:20
      CMake: 3.10.2
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: n/a
 Build type: release

Configuration
       File: /root/.taskrc (found), 1546 bytes, mode 100644
       Data: /root/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
     Server: 
      Trust: strict
Certificate: , not readable, 0 bytes
        Key: , not readable, 0 bytes
    Ciphers: NORMAL
      Creds: 

Hooks
     System: Enabled
   Location: /root/.task/hooks
             (-none-)

Tests
      $TERM: xterm (152x42)
       Dups: Scanned 1 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 1 tasks for broken references:
             No broken references found
```

Forgive all the `root`... this is in an Ubuntu 18.04 docker container.

Compiled with `cmake -DCMAKE_BUILD_TYPE=release -DENABLE_SYNC=OFF .`

Packages installed in the container: `build-essential`, `cmake, `uuid-dev`, `nano` (to quickly edit my `~/.taskrc` in the container).

**Comment by tbabej** [comment]
This is present on 2.6.0, the underlying issue appears to be that the big integers are overflowing the 32-bit signed int max (2,147,483,647):

    [root@test ~]# task add long uda rtm:123475873462
    Created task 1.
    [root@test ~]# task add short uda rtm:1234
    Created task 2.
    [root@test ~]# task
    [task next]
    
    ID Age  Description Urg  RTM
     1 8min long uda       0 -1078178122
     2 8min short uda      0        1234

**Comment by tbabej** [comment]
Additionally, numeric UDA values between 315532800 and  2147483647 are parsed as datetime epoch values (corresponding to 1980-01-01 and 2038-01-19 epoch times):

```
ID  Age  Description            Urg  RTM
 4  6min int max plus 2         0         -2147483647
 5  5min int max minus 1        0 2038-01-19T03:14:06
11  3min value of 100000000     0           100000000
12  3min value of 200000000     0           200000000
22 42s   value of 316000000     0 1980-01-06T09:46:40
```

**Comment by tbabej** [comment]
Fixed in 2.6.0.

**Comment by tbabej** [comment]
Warnings for overflow still need to be added (see #2379).
